src/dns/resolve.rs

```diff
@@ -53,7 +53,7 @@ impl FromStr for Name {
 
 #[derive(Clone)]
 pub(crate) struct DynResolver {
-    resolver: Arc<dyn Resolve>,
+    pub(crate) resolver: Arc<dyn Resolve + Sync>,
```

Comment:
IMHO you could also put the `Send` bound as well

```suggestion
    pub(crate) resolver: Arc<dyn Resolve + Send + Sync>,
```

Reply:
I noticed, that both bound are not necessary here as `Resolve` trait requires `Send` and `Sync` to be implementet. I removed them.